pkg/runtime/runtime.go

```diff
@@ -712,3 +749,28 @@ func (m *MosnRuntime) initCustomComponents(kind2factorys map[string][]*custom.Co
 	}
 	return nil
 }
+
+func (m *MosnRuntime) initComponentInject(comp interface{}, config *refconfig.ComponentRefConfig) error {
+	if _, ok := comp.(common.SetComponent); ok {
+		configRef, err := m.Injector.InjectConfigStoreRef(config)
+		if err != nil {
+			return err
+		}
+		if configRef != nil {
+			if err = comp.(common.SetComponent).SetConfigStore(configRef); err != nil {
```

Comment:
You can reuse the type assertion result
![image](https://user-images.githubusercontent.com/26001097/193156008-e4fb67e1-9eb9-4d87-bad0-6708bf79376d.png)